CareKitStore/CareKitStoreTests/OCKStore/TestStore+Sync.swift

```diff
@@ -267,6 +267,47 @@ class TestStoreSync: XCTestCase {
         XCTAssert(localOutcomes == remoteOutcomes)
         XCTAssert(localOutcomes.count == 1)
     }
+    
+    func testTombstoningOutcomePushedToRemote() throws {
+        let dummy2 = DummyEndpoint2()
+        let testStore = OCKStore(name: "test", type: .inMemory, remote: dummy2)
+        dummy2.automaticallySynchronizes = false
+        
+        let schedule = OCKSchedule.mealTimesEachDay(start: Date(), end: nil)
+        let task = try testStore.addTaskAndWait(OCKTask(id: "A", title: "A", carePlanUUID: nil, schedule: schedule))
+        let taskUUID = try task.getUUID()
+
+        let outcome = try testStore.addOutcomeAndWait(OCKOutcome(taskUUID: taskUUID, taskOccurrenceIndex: 0, values: [OCKOutcomeValue("test")]))
+        XCTAssertNoThrow(try testStore.syncAndWait()) //Sync original outcome
+        
+        try testStore.deleteOutcomeAndWait(outcome)
+        XCTAssertNoThrow(try testStore.syncAndWait()) //Sync tombstoned outcome
+        let synchedEntities = dummy2.entitiesPushed
+        XCTAssert(synchedEntities.count == 2)
+        
+        let tombstonedOutcomes = synchedEntities.compactMap{
+            entity -> OCKOutcome? in
+            switch entity{
+            case .outcome(let outcome):
+                return outcome
+            default:
+                return nil
+            }
+        }
+        XCTAssert(tombstonedOutcomes.count == 2)
+        
+        guard let tombstonedWithSameUUID = tombstonedOutcomes.filter({$0.uuid == outcome.uuid}).first else{
+            throw OCKStoreError.invalidValue(reason: "Filter doesn't contain UUID")
```

Comment:
Wasn’t sure what to throw here, if there’s a suggested way, I can change

Reply:
The error seems fine, especially since it's just a unit test.
Consider preferring `.first(where: { try $0.getUUID() == outcomeUUID })` over `.filter({ ... }).first`.

Reply:
Sure, I'll make this change

Reply:
Fixed